### PR TITLE
feat: allow for decimals to be used as input types for UDFs

### PR DESCRIPTION
**Pull request by agavra**
### Description 
This change makes it so that UDFs accept decimal types, but they still cannot return decimal types. To make this change, we treat all decimals as the same schema within the UDF context (which is OK because they all map to `BigDecimal`)

### Testing done 
- Unit Tests
- QTT tests (for `floor` and moved `floor` to be a `@Udf` instead of `Kudf`)

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-common/src/main/java/io/confluent/ksql/function/UdfIndex.java [hunk 1]
```diff
@@ -316,7 +318,6 @@ boolean accepts(final Schema argument, final Map<Schema, Schema> reservedGeneric
       return Objects.equals(type, argument.type())
           && CUSTOM_SCHEMA_EQ.getOrDefault(type, (a, b) -> true).test(schema, argument)
           && Objects.equals(schema.version(), argument.version())
-          && Objects.equals(schema.parameters(), argument.parameters())
```
Comment (vcrfxia) [inline ksql-common/src/main/java/io/confluent/ksql/function/UdfIndex.java]: Is it only the decimal type that may contain parameters in their schemas? (Trying to understand whether removing this portion of the check is safe.)

File: ksql-common/src/main/java/io/confluent/ksql/schema/connect/SqlSchemaFormatter.java [hunk 2]
```diff
@@ -126,7 +126,7 @@ protected String visitBytes(final Schema schema) {
           + DecimalUtil.scale(schema) + ")";
     }
 
-    throw new KsqlException("Cannot format bytes type: " + schema);
+    return "BYTES";
```
Comment (vcrfxia) [inline ksql-common/src/main/java/io/confluent/ksql/schema/connect/SqlSchemaFormatter.java]: It's still the case that the only bytes type we currently support is Decimals, right? I'm having trouble understanding why this change is necessary (besides the pretty error message in UdfIndexTest#shouldNotFindArbitraryBytesTypes`).

File: ksql-engine/src/main/java/io/confluent/ksql/function/UdfUtil.java [hunk 3]
```diff
@@ -93,7 +99,9 @@ static Schema getSchemaFromType(final Type type, final String name, final String
       schema = GenericsUtil.generic(((TypeVariable) type).getName());
     } else {
       schema = typeToSchema.getOrDefault(type, () -> handleParametrizedType(type)).get();
-      schema.name(name);
+      if (schema.name() == null) {
```
Comment (vcrfxia) [inline ksql-engine/src/main/java/io/confluent/ksql/function/UdfUtil.java]: Why is this change necessary? (Why would the schema already have a name at this point?)

File: ksql-common/src/main/java/io/confluent/ksql/function/UdfIndex.java (hunk 1 above)
Comment (agavra) [reply ksql-common/src/main/java/io/confluent/ksql/function/UdfIndex.java]: at the moment yes - it's the only one that we use that has parameters. I think in general it makes sense to defer the parameter to check to the `CUSTOM_SCHEMA_EQ` instead of checking that they are the same.

File: ksql-common/src/main/java/io/confluent/ksql/schema/connect/SqlSchemaFormatter.java (hunk 2 above)
Comment (agavra) [reply ksql-common/src/main/java/io/confluent/ksql/schema/connect/SqlSchemaFormatter.java]: just the pretty error message - if you can't find a UDF we shouldn't throw "Cannot format bytes type". The other BYTES type that we support is also the way we represent Generics.

File: ksql-engine/src/main/java/io/confluent/ksql/function/UdfUtil.java (hunk 3 above)
Comment (agavra) [reply ksql-engine/src/main/java/io/confluent/ksql/function/UdfUtil.java]: this is something that I feel like we should change in the long run, but decimals are defined by their schema name (they all have `org.apache.kafka.connect.data.Decimal` as their schema name). we should not be using the schema name to define the parameter name, but that's out of scope

Comment (vcrfxia) [reply ksql-engine/src/main/java/io/confluent/ksql/function/UdfUtil.java]: Ah, my bad. I even followed the Decimal schema builder code through but somehow missed it was setting the name. Thanks for the explanations!
